Userland/Services/WindowServer/Menu.cpp

```diff
@@ -692,12 +692,21 @@ void Menu::set_visible(bool visible)
         m_client->async_menu_visibility_did_change(m_menu_id, visible);
 }
 
+void Menu::update_alt_shortcuts_for_items() {
+    m_alt_shortcut_character_to_item_indices.clear();
+    int i = 0;
+    for (auto& item : m_items) {
+        if (auto alt_shortcut = find_ampersand_shortcut_character(item.text())) {
+            m_alt_shortcut_character_to_item_indices.ensure(to_ascii_lowercase(alt_shortcut)).append(i);
+        }
+        ++i;
+    }
+}
+
 void Menu::add_item(NonnullOwnPtr<MenuItem> item)
 {
-    if (auto alt_shortcut = find_ampersand_shortcut_character(item->text())) {
-        m_alt_shortcut_character_to_item_indices.ensure(to_ascii_lowercase(alt_shortcut)).append(m_items.size());
-    }
     m_items.append(move(item));
+    update_alt_shortcuts_for_items();
```

Comment:
Recalculating all the shortcuts every time an item is added feels a little awkward, in cases like this:

```c++
menu.add_item(an_item);
menu.add_item(another_item);
menu.add_item(yet_another_item);
menu.add_item(something);
menu.add_item(i_am_running_out_of_item_names);
```

Then, we're going through and updating the shortcuts 5 times. But then again, even with 20 items in a menu it's not that much work. It's fine, I'm just being paranoid. :sweat_smile:

Reply:
I opted for recalculating the shortcuts here rather than leaving the existing code in place, which AFAICT allowed assigning the same alt shortcut to multiple items. This seemed a bit shaky to me. :man_shrugging:

Reply:
Oh nice catch. Yeah, correctness is always more important.